### PR TITLE
Fix test framework for eksa version

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/controller/clusters/status.go [hunk 1]
```diff
@@ -22,6 +22,11 @@ func UpdateClusterStatusForControlPlane(ctx context.Context, client client.Clien
 		return errors.Wrapf(err, "getting kubeadmcontrolplane")
 	}
 
+	if kcp == nil {
```
Comment (g-gaston) [inline pkg/controller/clusters/status.go:25]: why this change? it's difficult to understand the changes without any context

Comment (tatlat) [reply pkg/controller/clusters/status.go:25]: it caused a nil pointer dereference when deleting a workload cluster in TestVSphereUpgradeKubernetes123to124UbuntuWorkloadClusterAPI in updateControlPlaneReadyCondition(cluster, kcp)

Comment (cxbrowne1207) [reply pkg/controller/clusters/status.go:25]: I'd vote to just return nil here instead of also setting the condition then. The ControlPlaneInitialized condition should not be updated again after being "True". This will update the value back to "False"

Comment (g-gaston) [reply pkg/controller/clusters/status.go:25]: is it easier to just add this check to `updateControlPlaneReadyCondition`?

Comment (tatlat) [reply pkg/controller/clusters/status.go:25]: yeah we can do that instead

Comment (cxbrowne1207) [reply pkg/controller/clusters/status.go:25]: yeah, i think that solution is better

**Comment by g-gaston** [comment]
/lgtm
Next time please add some description to the PR to help the reviewers build context :)
